budget-manager/src/modals/add-spending/add-spending-dialog.html

```diff
@@ -0,0 +1,29 @@
+<form>
+    <h1 mat-dialog-title>Add Spending</h1>
+    <div mat-dialog-content [formGroup]="form">
+        <mat-form-field>
+            <mat-label>Value</mat-label>
+            <input type="number" matInput formControlName="value" />
+        </mat-form-field>
+        <mat-form-field>
+            <mat-label>Name</mat-label>
+            <input type="text" matInput formControlName="name" />
+        </mat-form-field>
+        <mat-form-field appearance="fill">
+            <mat-label>Category</mat-label>
+            <mat-select formControlName="category">
+                <mat-option *ngFor="let category of data" value="{{category.name}}">{{category.name}}</mat-option>
+            </mat-select>
+        </mat-form-field>
+        <mat-form-field>
+            <mat-label>Choose a date</mat-label>
+            <input matInput [matDatepicker]="picker" formControlName="date">
+            <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+            <mat-datepicker #picker></mat-datepicker>
+        </mat-form-field>
+    </div>
+    <div mat-dialog-actions>
+        <button type="button" mat-button (click)="onCancel()">Cancel</button>
+        <button type="submit" mat-button (click)="onSubmit()" [disabled]="!form.valid">Submit</button>
```

Comment:
We've discussed this
```suggestion
        <button mat-button (click)="onSubmit()" [disabled]="!form.valid">Submit</button>
```

Reply:
fixed